web/pgadmin/tools/sqleditor/static/js/components/QueryToolDataGrid/index.jsx

```diff
@@ -92,13 +99,39 @@ const StyledPgReactDataGrid = styled(PgReactDataGrid)(({theme})=>({
 export const RowInfoContext = React.createContext();
 export const DataGridExtrasContext = React.createContext();
 
-function getCopyShortcutHandler(handleCopy) {
+function getShortcutKeyHandler({handleCopy, handleSelectAll}) {
   return (e)=>{
-    if((e.ctrlKey || e.metaKey) && e.key !== 'Control' && e.keyCode == 67) {
+    if(handleCopy && ((e.metaKey && isMac) || (e.ctrlKey && !isMac)) && e.key !== 'Control' && e.keyCode == 67) {
       e.preventDefault();
       handleCopy();
     }
+    if (handleSelectAll && ((e.metaKey && isMac) || (e.ctrlKey && !isMac)) && e.key === 'a') {
+      e.preventDefault();
+      handleSelectAll();
+    }
+  };
+}
+
+function handleShortcutKeyDown({
```

Comment:
there should only be one handler. don't make it complex by adding multiple layers of callbacks. I'm finding it very difficult to understand what is the code doing. There are no comments as well.

Reply:
Thanks, I will optimize it.